Restructure for DX (Developer Experience)

consideRatio opened the pull request:
I hope that the following changes can make the content of the repo more understandable for anyone working with the source code.

I recommend reviewing the PR commit by commit.

Ping @manics since you have done a lot of great work with related content!

---

Closes #1176

minrk commented on cd/publish-chart.sh:
```diff
@@ -0,0 +1,23 @@
+# Activate logging of bash commands now that the sensitive stuff is done
+set -x
+
+# As chartpress utilizes git to push to our Helm chart repository, we configure
```
Minor English tip: approximately every time you might use "utilize", "use" is probably a better choice.

consideRatio commented:
@minrk thanks for the review! Since your last review:
- I moved the minikube-config.yaml and minikube-netpol.yaml back to the root folder
- I moved publish_chart.sh and id_rsa.enc from the new cd folder to the ci folder instead.
- I reworded English: from utilize to use

minrk commented:
Excellent, thanks!
